### PR TITLE
Developed application id generator.

### PR DESCRIPTION
**Pull request by yashodgayashan**
Process of generating the application ID.
* First part consists of application code. _tree removing form -> trf_
* Application created date.
* Application count of particular application type.

Ex - trf-20200516-6

### CODE REVIEW

File: core/src/env_platform/db_handler.bal [hunk 1]
```diff
@@ -13,21 +13,23 @@ mongodb:ClientConfig mongoConfig = {
...
 mongodb:Database mongoDatabase = check mongoClient->getDatabase("EnvironmentPlatform");
 mongodb:Collection applicationCollection = check mongoDatabase->getCollection("applications");
+mongodb:Collection usersCollection = check mongoDatabase->getCollection("users");
 
 # The `saveApplication` function will save the application to the applications collection in the database.
 # 
 # + form - Form containing the tree removal data.
-# + return - Returns true if the application is saved, error if there is a mongodb:DatabaseError.
+# + return - Returns true if the application is saved, error if there is a mongodb:DatabaseError or error 
+# generated when the applicationId is generated.
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:22]:
```suggestion
# + return - Returns true if the application is saved, error if there is a mongodb:DatabaseError or  
# there's an error while generating the applicationId.
```

File: core/src/env_platform/db_handler.bal [hunk 2]
```diff
@@ -160,3 +162,11 @@ function updateApplication(TreeRemovalForm form, string applicationId) returns b
         return updated;
     }
 }
+
+# The `getApplicationCountByTitle` function will return the number of application for the given application type.
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:166]:
```suggestion
# The `getApplicationCountByTitle` function will return the number of application for a given application type.
```

File: core/src/env_platform/utility.bal [hunk 3]
```diff
@@ -84,3 +84,36 @@ function constructApplication(TreeRemovalForm form) returns map<json> {
         "treeInformation": extractTreeInformationAsJSONArray(form.treeInformation)
     };
 }
+
+# The `generateApplicationId` function will return a unique Id for a given application type. 
+# The format is <applicationCode>-<createdDate>-<applicationNumber>.
+# 
+# + createdDate - Created date of the application.
+# + applicationType -  Application type of the application.
+# + return - Returns a unique application Id or Corresponding error.
```
Comment (deshankoswatte) [inline core/src/env_platform/utility.bal:93]:
```suggestion
# + return - Returns a unique application Id or corresponding error.
```

File: core/src/env_platform/utility.bal [hunk 4]
```diff
@@ -84,3 +84,36 @@ function constructApplication(TreeRemovalForm form) returns map<json> {
...
+    time:Time timeCreated = check time:createTime(createdDate.year, createdDate.month, createdDate.day, 0, 0, 0, 0, "Asia/Colombo");
+    string customTimeString = check time:format(timeCreated, "yyyyMMdd");
+
+    // Get application count
+    int applicationCount = (check getApplicationCountByTitle(applicationType) + 1);
+    return applicationCode + "-" + customTimeString + "-" + applicationCount.toString();
+}
+
+
+# The `getApplicationCode` function will return the corresponding application code for the given application type.
```
Comment (deshankoswatte) [inline core/src/env_platform/utility.bal:109]:
```suggestion
# The `getApplicationCode` function will return the corresponding application code for a given application type.
```

**Comment by yashodgayashan** [comment]
@deshankoswatte resolved requested changes.

File: core/src/env_platform/db_handler.bal [hunk 5]
```diff
@@ -160,3 +162,11 @@ function updateApplication(TreeRemovalForm form, string applicationId) returns b
         return updated;
     }
 }
+
+# The `getApplicationCountByTitle` function will return the number of application for a given application type.
+# 
+# + applicationType - Type of the application.
+# + return - This function will return either number of application with the given application type or mongodb:DatabaseError.
```
Comment (sanjulamadurapperuma) [inline core/src/env_platform/db_handler.bal:169]:
```suggestion
# + return - This function will return either number of application with the given application type or 
# error if there is a mongodb:DatabaseError.
```

File: core/src/env_platform/utility.bal [hunk 6]
```diff
@@ -84,3 +84,36 @@ function constructApplication(TreeRemovalForm form) returns map<json> {
...
+# 
+# + createdDate - Created date of the application.
+# + applicationType -  Application type of the application.
+# + return - Returns a unique application Id or corresponding error.
+function generateApplicationId(Date createdDate, string applicationType) returns string|error {
+
+    // Get the application code
+    string applicationCode = check getApplicationCode(applicationType);
+
+    // Construct the date to a string
```
Comment (sanjulamadurapperuma) [inline core/src/env_platform/utility.bal:99]:
```suggestion
    // Convert the createdDate to a formatted string
```

**Comment by yashodgayashan** [comment]
@sanjulamadurapperuma resolved requested changes. @vikumkbv please review and accept.

**Comment by deshankoswatte** [comment]
@sanjulamadurapperuma Please approve an merge.

**Pull request merged**